rust-src/bulletproofs/benches/smp_bench.rs

```diff
@@ -0,0 +1,330 @@
+#[macro_use]
+extern crate criterion;
+
+use bulletproofs::{
+    inner_product_proof::verify_scalars,
+    set_membership_proof::*,
+    utils::{get_set_vector, z_vec, Generators},
+};
+use criterion::{BenchmarkId, Criterion};
+use curve_arithmetic::*;
+use ff::Field;
+use pairing::bls12_381::G1;
+use pedersen_scheme::{Commitment, CommitmentKey, Randomness};
+use rand::*;
+use random_oracle::RandomOracle;
+use std::time::Duration;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum UltraVerificationError {
+    /// The length of G_H was less than |S|, which is too small
+    NotEnoughGenerators,
+    /// Could not compute the inner product scalars
+    IpScalarCheckFailed,
+    /// Could not invert the given y
+    DivisionError,
+    /// Verification sum non-zero
+    VerificationSumNonZero,
+}
+
+#[allow(non_snake_case)]
+pub fn verify_ultra_efficient<C: Curve, R: Rng>(
```

Comment:
Can you please provide context for why we are doing this, why this is in the bench file, and what the conclusions are.

Essentially putting in this file what was discussed at the meeting last week.

Reply:
Removed because the normal verify is equally efficient.